liblangutil/SemVerHandler.cpp

```diff
@@ -250,6 +252,18 @@ SemVerMatchExpression::MatchComponent SemVerMatchExpressionParser::parseMatchCom
 		else
 			break;
 	}
+
+	// Validate that the parsed version parts are either a single string literal or multiple bare tokens,
+	// i.e. "1.2.3" or 1.2.3 but not 1."2.3", "1".2.3 or 1"."2.3.
+	auto const partsEndPos = m_pos; // Points *after* the last version part
+	for (auto i = partsStartPos; i < partsEndPos; ++i)
+	{
+		if (m_tokens[i] == Token::StringLiteral && partsStartPos != partsEndPos - 1)
+		{
+			solThrow(SemVerError, "String literals are only allowed as the only component in a version pragma.");
+		}
```

Comment:
Also a side note - our style is to not use scoping braces `{}` in conditional/loop blocks if they're one liners, i.e., this if block would turn into:

```suggestion
		if (m_tokens[i] == Token::StringLiteral && partsStartPos != partsEndPos - 1)
			solThrow(SemVerError, "String literals are only allowed as the only component in a version pragma.");
```